main.go

```diff
@@ -3,63 +3,106 @@ package main
 import (
 	"encoding/json"
 	"fmt"
-	"github.com/equinor/radix-cost-allocation/models"
-	"log"
-	"os"
+
 	"time"
 
+	"github.com/equinor/radix-cost-allocation/clients"
+	"github.com/equinor/radix-cost-allocation/models"
+	"github.com/pkg/errors"
+	"github.com/robfig/cron"
+	log "github.com/sirupsen/logrus"
+	"github.com/vrischmann/envconfig"
+	"golang.org/x/sync/semaphore"
+
 	_ "github.com/denisenkom/go-mssqldb"
 )
 
-// todo! create write only connection string? dont need read/admin access
-const port = 1433
+var Conf struct {
+	PrometheusAPI string
+	CronSchedule  string `envconfig:"default=0 * * * *"`
+	SQL           struct {
+		Server   string
+		Database string `envconfig:"default=sqldb-radix-cost-allocation"`
+		User     string
+		Password string
+		Port     int `envconfig:"default=1433"`
+	}
+}
+
+var sem = semaphore.NewWeighted(1)
 
 func main() {
-	promClient := PrometheusClient{Address: os.Getenv("PROMETHEUS_API")}
-	sqlClient := NewSQLClient(os.Getenv("SQL_SERVER"), os.Getenv("SQL_DATABASE"), port, os.Getenv("SQL_USER"), os.Getenv("SQL_PASSWORD"))
+	if err := envconfig.Init(&Conf); err != nil {
+		log.Fatal(err)
+	}
+
+	promClient := clients.PrometheusClient{Address: Conf.PrometheusAPI}
+	sqlClient, err := clients.NewSQLClient(Conf.SQL.Server, Conf.SQL.Database, Conf.SQL.Port, Conf.SQL.User, Conf.SQL.Password)
+	if err != nil {
+		log.Fatal(err)
+		return
+	}
+	defer sqlClient.Close()
 
-	moveResourceRequestsFromPrometheusToSQLDB(promClient, sqlClient)
 	// printCostBetweenDates(time.Now().UTC().AddDate(0, 0, -3), time.Now().UTC(), promClient, sqlClient)
 
-	sqlClient.Close()
+	initAndRunDataCollector(promClient, sqlClient)
+}
+
+func initAndRunDataCollector(promClient clients.PrometheusClient, sqlClient clients.SQLClient) {
+	c := cron.New()
+	c.AddFunc(Conf.CronSchedule, func() {
+		if !sem.TryAcquire(1) {
+			return
+		}
+		defer sem.Release(1)
+
+		if err := moveResourceRequestsFromPrometheusToSQLDB(promClient, sqlClient); err != nil {
+			log.Error(err)
+		}
+	})
+
+	c.Run()
 }
 
-func printCostBetweenDates(from, to time.Time, promClient PrometheusClient, sqlClient SQLClient) {
+func printCostBetweenDates(from, to time.Time, promClient clients.PrometheusClient, sqlClient clients.SQLClient) error {
```

Comment:
Not used?

Reply:
Probably just for debugging purposes. Removed.